WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderCommentListActivity.java

```diff
@@ -189,43 +183,38 @@ public void onClick(View v) {
         }
     };
 
+    // to do a complete refresh we need to get updated post and new comments
     private void updatePostAndComments() {
-        //to do a complete refresh we need to get updated post and new comments
         ReaderPostActions.updatePost(mPost, new ReaderActions.UpdateResultListener() {
             @Override
             public void onUpdateResult(ReaderActions.UpdateResult result) {
-                if (isFinishing()) {
-                    return;
-                }
-
-                if (result.isNewOrChanged()) {
-                    getCommentAdapter().setPost(mPost); //pass updated post to the adapter
-                    ReaderCommentTable.purgeCommentsForPost(mBlogId, mPostId); //clear all the previous comments
-                    updateComments(false, false); //load first page of comments
-                } else {
-                    setRefreshing(false);
+                if (!isFinishing() && result.isNewOrChanged()) {
+                    // get the updated post and pass it to the adapter
+                    ReaderPost post = ReaderPostTable.getBlogPost(mBlogId, mPostId, false);
+                    if (post != null) {
+                        getCommentAdapter().setPost(post);
+                        mPost = post;
+                    }
                 }
             }
         });
+
+        // clear all the previous comments, then load the first page of comments
+        ReaderCommentTable.purgeCommentsForPost(mBlogId, mPostId);
```

Comment:
Are we really sure we want to purge all the comments here before waiting for server response? 
If the network call fails the client is stuck with no comments at all. I just tried by making the call failing, and then rotated the device.